modules/reindex/src/javaRestTest/java/org/elasticsearch/index/reindex/remote/ReindexFromOldRemoteIT.java

```diff
@@ -92,8 +93,17 @@ private void oldEsTestCase(String portPropertyName, String requestsPerSecond) th
                 for (int i = 0; i < DOCS; i++) {
                     assertThat(result, containsString("\"_id\" : \"testdoc" + i + "\""));
                 }
+                success = true;
             } finally {
-                oldEs.performRequest(new Request("DELETE", "/test"));
+                try {
+                    oldEs.performRequest(new Request("DELETE", "/test"));
+                } catch (Exception deleteException) {
+                    logger.warn("Exception deleting index", deleteException);
+                    if (success) {
+                        // Since the test completed successfully, the delete should not fail
+                        throw deleteException;
```

Comment:
I think the comment above is slightly misleading as this branch would effectively throw exception.
May be reword to `Since the test completed successfully, the delete exception should not be swallowed if present`?